src/osversion/test/VersionTranslator_GTest.cpp

```diff
@@ -2418,3 +2418,51 @@ TEST_F(OSVersionFixture, update_3_5_0_to_3_5_1_UnitarySystemPerformanceMultispee
   EXPECT_EQ("autosize", eg2.getString(0, false).get());
   EXPECT_EQ(3.0, eg2.getDouble(1, false).get());
 }
+
+TEST_F(OSVersionFixture, update_3_5_0_to_3_5_1_VRF_Terminal_v350_osc) {
+  // In this one, the starting version is 3.5.0, so it already has Table:Lookup instead of Table:MultiVariableLookup
+  openstudio::path p = resourcesPath() / toPath("osversion/3_5_1/test_VRF_Terminal_v350.osc");
+  osversion::VersionTranslator vt;
+  boost::optional<model::Component> comp_ = vt.loadComponent(p);
+  ASSERT_TRUE(comp_) << "Failed to load Component " << p;
+
+  openstudio::path outPath = resourcesPath() / toPath("osversion/3_5_1/test_VRF_Terminal_v350_updated.osc");
+  comp_->save(outPath, true);
+
+  auto compData = comp_->componentData();
+  EXPECT_EQ(18, compData.numComponentObjects());
+  EXPECT_EQ("OS:ZoneHVAC:TerminalUnit:VariableRefrigerantFlow", compData.primaryComponentObject().iddObject().name());
+
+  Model model;
+  OptionalComponentData ocd = model.insertComponent(comp_.get());
+  ASSERT_TRUE(ocd);
+  EXPECT_EQ("OS:ZoneHVAC:TerminalUnit:VariableRefrigerantFlow", compData.primaryComponentObject().iddObject().name());
+  EXPECT_EQ(18, ocd->numComponentObjects());
+  EXPECT_EQ(19, model.numObjects());
+  EXPECT_EQ("OS:ZoneHVAC:TerminalUnit:VariableRefrigerantFlow", ocd->primaryComponentObject().iddObject().name());
+}
+
+TEST_F(OSVersionFixture, update_3_5_0_to_3_5_1_VRF_Terminal_v340_osc) {
+  // Here the starting version is 3.4.0, so it has only 12 objects to start with
+  openstudio::path p = resourcesPath() / toPath("osversion/3_5_1/test_VRF_Terminal_v340.osc");
+  osversion::VersionTranslator vt;
+  boost::optional<model::Component> comp_ = vt.loadComponent(p);
+  ASSERT_TRUE(comp_) << "Failed to load Component " << p;
```

Comment:
another test with https://github.com/NREL/openstudio-aedg-gem/blob/develop/lib/measures/zedgk_12_hvac_vrf_with_doas/resources/VRF_Terminal_old.osc that I did vt to 3.4.0 for speed (but still has Table:MultiVariableLookup) to make sure it works there too.